pennylane_lightning/core/src/simulators/lightning_qubit/measurements/ExpValFunctorsLQubit.hpp

```diff
@@ -0,0 +1,315 @@
+// Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the License);
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an AS IS BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file ExpValFunctorsQubit.hpp
+ * Define functors for in-place computation of expectation value of
+ * Identity, PauliX, PauliY, PauliZ, and Hadamard operators.
+ */
+
+#pragma once
+
+#include "BitUtil.hpp"
+#include "Util.hpp"
+
+/// @cond DEV
+namespace {
+using namespace Pennylane::Util;
+} // namespace
+/// @endcond
+
+namespace Pennylane::LightningQubit::Functors {
+/**
+ * @brief Functor for computing expectation value of Identity operator.
+ *
+ * This functor provides an in-place implementation for computing the
+ * expectation value of Identity operator for a given statevector.
+ *
+ * @tparam PrecisionT precision data type of the statevector.
+ */
+template <class PrecisionT> struct getExpectationValueIdentityFunctor {
```

Comment:
```suggestion
template <class PrecisionT> struct getExpectationValueIdentityFunctor {
```

Since we are now defining private members, we should make this a class :)

Reply:
I just changed the `struct` to `class`.